qiskit/pulse/builder.py

```diff
@@ -0,0 +1,1963 @@
+                        pulse.play(pulse.Constant(100, 1.0), d1)
+                        # this pulse will also start at t=0
+                        pulse.play(pulse.Constant(20, 1.0), d2)
+
+        assert (pulse_prog.ch_start_time(d1) ==
+                pulse_prog.ch_start_time(d2) ==
+                pulse_prog.ch_start_time(d1))
+
+    .. warning:: This will cause all scheduling directives within this context
+        to be ignored.
+    """
+    builder = _active_builder()
+    active_block = builder.block
+    transform_block = Schedule()
+    builder.set_active_block(transform_block)
+    try:
+        yield
+    finally:
+        builder._compile_lazy_circuit()
+        builder.set_active_block(active_block)
+        for _, instruction in transform_block.instructions:
+            append_instruction(instruction)
+
+
+@_transform_context(transforms.pad, mutate=True)
+def pad(*chs: channels.Channel) -> ContextManager[None]:  # pylint: disable=unused-argument
+    """Pad all availale timeslots with delays upon exiting context.
+
+    Args:
+        chs: Channels to pad with delays. Defaults to all channels in context
+            if none are supplied.
+
+    Example Usage:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+
+        with pulse.build() as pulse_prog:
+            with pulse.pad():
+                with pulse.align_right():
+                    # this pulse will start at t=0
+                    pulse.play(pulse.Constant(100, 1.0), d0)
+                    # this pulse will start at t=80
+                    # a delay will be inserted from t=0 to t=80
+                    pulse.play(pulse.Constant(20, 1.0), d1)
+        assert pulse_prog.ch_start_time(d0) == pulse_prog.ch_start_time(d1)
+        assert pulse_prog.ch_stop_time(d0) == pulse_prog.ch_stop_time(d1)
+    """
+
+
+@contextmanager
+def transpiler_settings(**settings) -> ContextManager[None]:
+    """Set the currently active transpiler settings for this context.
+
+    Example Usage:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+        from qiskit.test.mock import FakeOpenPulse2Q
+
+        backend = FakeOpenPulse2Q()
+
+        with pulse.build(backend):
+            print(pulse.active_transpiler_settings())
+            with pulse.transpiler_settings(optimization_level=3):
+                print(pulse.active_transpiler_settings())
+    """
+    builder = _active_builder()
+    curr_transpiler_settings = builder.transpiler_settings
+    builder.transpiler_settings = collections.ChainMap(
+        settings, curr_transpiler_settings)
+    try:
+        yield
+    finally:
+        builder.transpiler_settings = curr_transpiler_settings
+
+
+@contextmanager
+def circuit_scheduler_settings(**settings) -> ContextManager[None]:
+    """Set the currently active circuit scheduler settings for this context.
+
+    Example Usage:
```

Comment:
You're really thorough with documentation --- that's awesome!

Reply:
I really tried!!! It took a long time, Thanks for noticing 😄 .